src/libraries/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs

```diff
@@ -70,10 +70,15 @@ public override SecurityRuleSet SecurityRuleSet
             get { return UnderlyingAssembly.SecurityRuleSet; }
         }
 
+#if NETCOREAPP
+        [Obsolete]
+#endif
+#pragma warning disable IL3002 // Using member which has 'RequiresAssemblyFilesAttribute' can break functionality when embedded in a single-file app
```

Comment:
cc: @eerhardt

Reply:
These aren't correct, we should be annotating these with `[RequiresAssemblyFiles]` attributes, not `[Obsolete]`. For example:

https://github.com/dotnet/runtime/blob/04ad80aff37e2ef5eac0cc6d2a895ccf0f9d65a3/src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs#L84-L85

https://github.com/dotnet/runtime/blob/04ad80aff37e2ef5eac0cc6d2a895ccf0f9d65a3/src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs#L119-L120

For TFMs where RequiresAssemblyFilesAttribute isn't defined, you can just source link include them into the assembly, and they will be `internal`. Example:

https://github.com/dotnet/runtime/blob/5f15498d937653986821d636c04cd03c58d4bb4c/src/libraries/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel.csproj#L16

cc @agocke @tlakollo

Reply:
Shouldn't it be annotated with `Obsolete` *and* `RequiresAssemblyFiles`? I thought CodeBase is already obsolete.

Reply:
Looks like it is marked obsolete in the ref, but not the implementation… weird. That seems like a mistake IMO

Reply:
Thanks I forgot about the `RequiresAssemblyFiles` attribute... 🗡️